Extract CSS::None/CSS::Range into their own files to avoid cycle with calc

weinig opened the pull request:
#### d963cf33c1630e3db30df7dd51c9167d4c27768f
<pre>
Extract CSS::None/CSS::Range into their own files to avoid cycle with calc
<a href="https://bugs.webkit.org/show_bug.cgi?id=282955">https://bugs.webkit.org/show_bug.cgi?id=282955</a>

Reviewed by Fujii Hironori.

In order for CSSUnevaluatedCalc.h to include CSSCalcValue.h, a few types
needed to extracted into their own files.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSGradientValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/color/CSSColorDescriptors.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumerDefinitions.h:
* Source/WebCore/css/values/backgrounds/CSSBorderRadius.cpp:
* Source/WebCore/css/values/images/CSSGradient.cpp:
* Source/WebCore/css/values/primitives/CSSNone.cpp: Added.
* Source/WebCore/css/values/primitives/CSSNone.h: Added.
* Source/WebCore/css/values/primitives/CSSPosition.cpp:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericRange.h: Added.
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+EvaluateCalc.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+Serialization.cpp:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+Serialization.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes.h:
* Source/WebCore/css/values/primitives/CSSUnevaluatedCalc.cpp:
* Source/WebCore/css/values/primitives/CSSUnevaluatedCalc.h:
* Source/WebCore/css/values/shapes/CSSCircleFunction.cpp:
* Source/WebCore/css/values/shapes/CSSEllipseFunction.cpp:
* Source/WebCore/css/values/shapes/CSSInsetFunction.cpp:
* Source/WebCore/css/values/shapes/CSSPathFunction.cpp:
* Source/WebCore/css/values/shapes/CSSPolygonFunction.cpp:
* Source/WebCore/css/values/shapes/CSSRectFunction.cpp:
* Source/WebCore/css/values/shapes/CSSShapeFunction.cpp:
* Source/WebCore/css/values/shapes/CSSShapeFunction.h:
* Source/WebCore/css/values/shapes/CSSXywhFunction.cpp:
* Source/WebCore/style/values/StyleValueTypes.h:
* Source/WebCore/style/values/primitives/StyleNone.h: Added.
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Blending.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Conversions.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes.h:

Canonical link: <a href="https://commits.webkit.org/286531@main">https://commits.webkit.org/286531@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1362a7ed6e69e956b340220e7461fa1ecaeeb522

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76222 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55252 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29124 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80727 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27494 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/78338 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64396 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3548 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59770 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17910 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79289 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49671 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65463 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40115 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47067 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22949 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25816 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68199 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23283 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82187 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3594 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2345 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67989 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3748 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65433 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67300 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16786 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11261 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9374 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3542 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/6349 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3565 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/6994 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/5323 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (approved):
Thank you for working on this. Looks good to me if EWS is all happy green.

weinig committed:
Extract CSS::None/CSS::Range into their own files to avoid cycle with calc
https://bugs.webkit.org/show_bug.cgi?id=282955

Reviewed by Fujii Hironori.

In order for CSSUnevaluatedCalc.h to include CSSCalcValue.h, a few types
needed to extracted into their own files.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSGradientValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/color/CSSColorDescriptors.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumerDefinitions.h:
* Source/WebCore/css/values/backgrounds/CSSBorderRadius.cpp:
* Source/WebCore/css/values/images/CSSGradient.cpp:
* Source/WebCore/css/values/primitives/CSSNone.cpp: Added.
* Source/WebCore/css/values/primitives/CSSNone.h: Added.
* Source/WebCore/css/values/primitives/CSSPosition.cpp:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericRange.h: Added.
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+EvaluateCalc.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+Serialization.cpp:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes+Serialization.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes.h:
* Source/WebCore/css/values/primitives/CSSUnevaluatedCalc.cpp:
* Source/WebCore/css/values/primitives/CSSUnevaluatedCalc.h:
* Source/WebCore/css/values/shapes/CSSCircleFunction.cpp:
* Source/WebCore/css/values/shapes/CSSEllipseFunction.cpp:
* Source/WebCore/css/values/shapes/CSSInsetFunction.cpp:
* Source/WebCore/css/values/shapes/CSSPathFunction.cpp:
* Source/WebCore/css/values/shapes/CSSPolygonFunction.cpp:
* Source/WebCore/css/values/shapes/CSSRectFunction.cpp:
* Source/WebCore/css/values/shapes/CSSShapeFunction.cpp:
* Source/WebCore/css/values/shapes/CSSShapeFunction.h:
* Source/WebCore/css/values/shapes/CSSXywhFunction.cpp:
* Source/WebCore/style/values/StyleValueTypes.h:
* Source/WebCore/style/values/primitives/StyleNone.h: Added.
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Blending.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Conversions.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes.h:

Canonical link: https://commits.webkit.org/286531@main
